service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -845,22 +878,40 @@ safe_call("""${RUN_IPYTHON} profile create && echo "c.InteractiveShellApp.extens
 print("hail installed successfully.")
 
 EOF
+fi
 
-  # Fork the following into background process to wait for dataproc to finish
-  # installing optional components including jupyter and start the proxy gateway service.
-  # Then safely install hail.
-  "$(
-    while ! systemctl is-active --quiet ${GOOGLE_DATAPROC_COMPONENT_GATEWAY_SERVICE_NAME}; do
-      sleep 5
-      emit "Waiting for ${GOOGLE_DATAPROC_COMPONENT_GATEWAY_SERVICE_NAME} to start..."
-    done
+# Fork the following into background process to execute after dataproc finishes
+# setting up its optional components.
+#
+# Post dataproc setup tasks:
+# 1. Wait for the dataproc component gateway service to start.
```

Comment:
Nits - when referring to dataproc in comments, make it "Dataproc" and when referring to hail, make it "Hail", except when you are referring to binaries or usernames that are in fact all lower case.